### ISSUE TITLE
Evaluation memoization service

### ISSUE THREAD

**Issue by RaitoBezarius** [NixOS/nixos-org-configurations#355]
**Is your feature request related to a problem? Please describe.**

I am always frustrated whenever I build something that relies on listening to channel bumps and consuming evaluation results, I cannot find a single place where I could source them and would need to memoize them myself.

This is expensive on my weak web services.

**Describe the solution you'd like**

I would like an evaluation memoization service that could reuse Hydra evaluation results.

Something where I could:

(a) listen to new channel bumps
(b) ask for any evaluation result for any channel bump

If it does exist, I would like documentation to rely on this and best practices.

**Describe alternatives you've considered**

- Re-evaluate everything I care about, etc. Note that I build it in https://github.com/Nix-Security-WG/nix-security-tracker/blob/main/src/website/shared/listeners/nix_evaluation.py.
- Do nothing

**Additional context**

**Comment by zimbatm** [comment]
Meta comment: Ideally, we can get members of the community involved in the design and development of this service. And then have a good hand-over to the infra team for deployment. I don't know how well the infra team can scale if they also become responsible for R&D.

**Comment by RaitoBezarius** [comment]
> Meta comment: Ideally, we can get members of the community involved in the design and development of this service. And then have a good hand-over to the infra team for deployment. I don't know how well the infra team can scale if they also become responsible for R&D.

I agree, as I mention it, I developed similar stuff (linked in the body of the original issue) and I can develop a specialized tool for it, but it would make sense to reuse Hydra evaluation results for that and pass them on to the right tool. Because of that, it seems hard to me to put such an issue elsewhere than the infra issue tracker.

**Comment by JulienMalka** [comment]
@tomberek you wanted to do something similar if I remember correctly our discussion at OceanSprint. I personally am also interested for research purposes.
